zerver/views/custom_profile_fields.py

```diff
@@ -90,8 +102,18 @@ def update_user_custom_profile_data(
         except CustomProfileField.DoesNotExist:
             return json_error(_('Field id {id} not found.').format(id=field_id))
 
-        validator = CustomProfileField.FIELD_VALIDATORS[field.field_type]
-        result = validator('value[{}]'.format(field_id), item['value'])
+        validators = CustomProfileField.FIELD_VALIDATORS
+        extended_validators = CustomProfileField.EXTENDED_FIELD_VALIDATORS
+        field_type = field.field_type
+        value = item['value']
+        try:
+            validator = validators[field_type]
+            result = validator('value[{}]'.format(field_id), value)
+        except KeyError:
+            extended_validator = extended_validators[field_type]
+            field_data = field.field_data
+            result = extended_validator('value[{}]'.format(field_id), field_data, value)
```

Comment:
This `try/except`  code path feels messy; can we at least do `if field_type in validators` for making the decision?